src/uwtools/tests/test_cli.py

```diff
@@ -209,29 +215,71 @@ def test_dispatch_template_render_yaml(caplog):
 def test_main_fail(params):
     fnretval, status, quiet, verbose = params
     with patch.multiple(
-        cli, check_args=D, parse_args=D, dispatch_config=D, setup_logging=D
+        cli, _check_args=D, _parse_args=D, _dispatch_config=D, setup_logging=D
     ) as mocks:
         args = ns(mode="config", quiet=quiet, verbose=verbose)
-        mocks["parse_args"].return_value = args
-        mocks["check_args"].return_value = mocks["parse_args"]()
-        mocks["dispatch_config"].return_value = fnretval
+        mocks["_parse_args"].return_value = args
+        mocks["_check_args"].return_value = mocks["_parse_args"]()
+        mocks["_dispatch_config"].return_value = fnretval
         with raises(SystemExit) as e:
             cli.main()
         assert e.value.code == status
-        mocks["dispatch_config"].assert_called_once_with(args)
-        mocks["check_args"].assert_called_once_with(args)
-        mocks["setup_logging"].assert_called_once_with(quiet=quiet, verbose=verbose)
+        mocks["_dispatch_config"].assert_called_once_with(args)
+        mocks["_check_args"].assert_called_once_with(args)
+        mocks["setup_logging"].assert_called_with(quiet=quiet, verbose=verbose)
+
 
+def test_main_raises_exception(capsys):
+    msg = "Test failed intentionally"
+    with patch.object(cli, "_parse_args", side_effect=Exception(msg)):
+        with raises(SystemExit):
+            cli.main()
+    assert msg in capsys.readouterr().err
 
-def test_parse_args():
+
+def test__parse_args():
     raw_args = ["foo", "--bar", "88"]
     with patch.object(cli, "Parser") as Parser:
-        cli.parse_args(raw_args)
+        cli._parse_args(raw_args)
         Parser.assert_called_once()
         parser = Parser()
         parser.parse_args.assert_called_with(raw_args)
 
 
```

Comment:
Here and below, new tests for the now-optional (but sometimes required!) data-format CLI arguments.